llms/mlx_lm/server.py

```diff
@@ -234,7 +248,12 @@ def generate_response(
                 break
 
         text = _tokenizer.decode(tokens)
-        return response_creator(response_id, requested_model, prompt, tokens, text)
+        response = response_creator(response_id, requested_model, prompt, tokens, text)
+        try:
+            self.wfile.write(f"{response}\n\n".encode())
```

Comment:
@mzbac Responses the the handle_stream method send the json as a "data:" header. Is that how OpenAI does it? If so we'll need to add it here too. The OpenAI API reference doesn't mention if json is part of the body or data header, at least I couldn't find anything

Reply:
It's necessary. I would suggest that when you work on the server, it's better to set up integration with some clients so you can test out your changes. For example, set up chat-ui with mlx-lm.server https://github.com/mzbac/mlx-chat-ui

Reply:
So far I've been working to try to match the OpenAI API Reference, but I guess their documentation does have its limits. I do find it strange though that json data would be contained in a http header.

Reply:
What do you mean by "in header"? The stream response is being sent via SSE. https://developer.mozilla.org/en-US/docs/Web/API/Server-sent_events/Using_server-sent_events

Reply:
Ah that link helps, ty